service-api/module/Application/src/DWP/DwpApi/DwpApiService.php

```diff
@@ -0,0 +1,200 @@
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws DwpApiException
+     */
+    public function makeCitizenMatchRequest(
+        CitizenRequestDTO $citizenRequestDTO,
+    ): CitizenResponseDTO {
+        $this->authCount++;
+        try {
+            $response = $this->guzzleClient->request(
+                'POST',
+                $this->matchPath,
+                [
+                    'headers' => $this->makeHeaders(),
+                    'json' => $citizenRequestDTO->constructCitizenRequestBody()
+                ]
+            );
+            $responseArray = json_decode($response->getBody()->getContents(), true);
+        } catch (ClientException $clientException) {
+            if (
+                $clientException->getResponse()->getStatusCode() == Response::STATUS_CODE_401 &&
+                $this->authCount < 2
+            ) {
+                $this->authApiService->authenticate();
+                $responseArray = $this->makeCitizenMatchRequest($citizenRequestDTO);
+            } else {
+                $response = $clientException->getResponse();
+                $responseBodyAsString = $response->getBody()->getContents();
+                $this->logger->error('GuzzleDwpApiException: ' . $responseBodyAsString);
+                throw $clientException;
+            }
+        } catch (\Exception $exception) {
+            throw new DwpApiException($exception->getMessage());
+        }
+        return new CitizenResponseDTO(
+            $responseArray
+        );
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws DwpApiException
+     */
+    public function makeCitizenDetailsRequest(
+        DetailsRequestDTO $detailsRequestDTO,
+        string $nino
+    ): DetailsResponseDTO {
+        $this->authCount++;
+        $responseArray = [];
+        try {
+            $uri = $this->detailsPath . $detailsRequestDTO->id();
+            $response = $this->guzzleClient->request(
+                'GET',
+                $uri,
+                [
+                    'headers' => $this->makeHeaders(['Access-Level => 4']),
+                ]
+            );
+            $responseArray = json_decode($response->getBody()->getContents(), true);
+            $this->logger->info("DWP_DETAILS_RESPONSE: " . json_encode($responseArray, JSON_THROW_ON_ERROR));
+        } catch (ClientException $clientException) {
+            if (
+                $clientException->getResponse()->getStatusCode() == Response::STATUS_CODE_401 &&
+                $this->authCount < 2
+            ) {
+                $this->authApiService->authenticate();
+                $this->makeCitizenDetailsRequest($detailsRequestDTO, $nino);
+            } else {
+                $response = $clientException->getResponse();
+                $responseBodyAsString = $response->getBody()->getContents();
+                $this->logger->error('GuzzleDwpApiException: ' . $responseBodyAsString);
```

Comment:
Is the body always an error string? Depending on what is in the error contents, might also be potential for PII leakage here too.